fe/fe-core/src/main/java/com/starrocks/sql/ast/SelectRelation.java

```diff
@@ -185,8 +218,112 @@ public void clearOrder() {
         return columnReferences;
     }
 
+    @Override
     public <R, C> R accept(AstVisitor<R, C> visitor, C context) {
         return visitor.visitSelect(this, context);
     }
+
+    public void setOutputExpr(List<Expr> outputExpr) {
```

Comment:
how about buld relation with a relation builder, so we it can be immutable

Reply:
no, relation is mutable.Because you need to analyze on it